src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1377,6 +1378,14 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
                     case 'flock' === $storeDsn:
                         $storeDefinition = new Reference('lock.store.flock');
                         break;
+                    case 0 === strpos($storeDsn, 'flock://'):
+                        $flockPath = substr($storeDsn, 8);
+
+                        $storeDefinitionId = '.lock.flock.store.'.$container->hash($storeDsn);
+                        $container->register($storeDefinitionId, FlockStore::class)->addArgument($flockPath);
+
+                        $storeDefinition = new Reference($storeDefinitionId);
```

Comment:
Do we need the reference? I mean if I don't miss anything, we do not reuse it anyway, so something like this should work the same:

```php
$flockPath = substr($storeDsn, 8);
$storeDefinitionId = '.lock.flock.store.'.$container->hash($storeDsn);
$storeDefinition = $container->register($storeDefinitionId, FlockStore::class)->addArgument($flockPath);
```

Reply:
Same can be said for both 'flock' and 'semaphore' cases. But I'm not sure that this PR is correct place for such refactoring. Besides '$storeDefinitions' array should contain 'Reference' objects